src/fparser/two/Fortran2003.py

```diff
@@ -7947,14 +7983,26 @@ class Nonlabel_Do_Stmt(StmtBase, WORDClsBase):  # pylint: disable=invalid-name
     subclass_names = []
     use_names = ["Do_Construct_Name", "Loop_Control"]
 
-    @staticmethod
-    def match(string):
+    @classmethod
+    def match(cls, string):
         """
         :param str string: Fortran code to check for a match
         :return: code line matching the nonlabeled "DO" statement
         :rtype: string
```

Comment:
Not yours but s/string/str/ please.

Reply:
Hopefully you meant for the return type as that is what I've changed.